gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/linker/EncassLinker.java

```diff
@@ -27,6 +33,18 @@ public void link(Encass encass, Bundle bundle, Bundle targetBundle) {
         if (policy == null) {
             throw new LinkerException("Could not find policy for Encapsulated Assertion: " + encass.getName() + ". Policy ID: " + encass.getPolicyId());
         }
+        Element encassPortalIntegrationElement = null;
+        Element encassPortalIntegrationEnabledElement = null;
+        try {
+            encassPortalIntegrationElement = getSingleElement(policy.getPolicyDocument(), API_PORTAL_ENCASS_INTEGRATION);
+        } catch (DocumentParseException ex) {
+            // do nothing
+        }
+        if (encassPortalIntegrationElement != null) {
```

Comment:
Can you please follow the pattern where we simplify the policy xmls using PolicyAssertionSimplifier implementations.

Reply:
@graju256 , Initially I thought of doing it as part of EncapsulatedAssertionSimplifier, but an Encass can exist without being part of any policy.

Reply:
> @graju256 , Initially I thought of doing it as part of EncapsulatedAssertionSimplifier, but an Encass can exist without being part of any policy.

You can search the bundle if any encas is being defined by this policy or not. if YES, continue removing the assertion and update the all the encases about it.

Reply:
> > @graju256 , Initially I thought of doing it as part of EncapsulatedAssertionSimplifier, but an Encass can exist without being part of any policy.
> 
> You can search the bundle if any encas is being defined by this policy or not. if YES, continue removing the assertion and update the all the encases about it.

Ignore my comment if you see any implementation difficulties. Moreover, I looked at the linker sequence as well: They were applied in a acceptable sequence like below:
 ENCAS, SERVICE, ....

Reply:
Sorry, I didn't read your comment properly. I'm thinking to have ApiPortalEncasIntegrationAssertionSimplifier. 
At lease, you can move the logic into a separate class say: **EncasPolicyXmlSimplifier** as a singleton.

Reply:
@graju256 will create a ApiPortalEncasIntegrationAssertionSimplifier class

Reply:
@graju256 few challenges with ApiPortalEncasIntegrationAssertionSimplifier class, moved the logic to separate singleton class EncassPolicyXMLSimplifier